model/src/test/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModelSpec.scala

```diff
@@ -0,0 +1,94 @@
+package org.broadinstitute.dsde.rawls.model
+
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.scalatest.{FreeSpec, Matchers}
+
+class WorkspaceModelSpec extends FreeSpec with Matchers {
+
+  "MethodRepoMethod" - {
+    "Can convert to a method repo URI" - {
+      "For valid methods" - {
+        val method = MethodRepoMethod("test-namespace", "test-name", 555)
+
+        "for Agora" in {
+          assertResult("agora://test-namespace/test-name/555") {
+            method.asAgoraMethodUrl
+          }
+        }
+
+        "for some other method repo" in {
+          assertResult("marks-methods-mart://test-namespace/test-name/555") {
+            method.asMethodUrlForRepo("marks-methods-mart")
+          }
+        }
```

Comment:
I think we should have a whitelist (via constants/enums) of the repos we support, and not allow arbitrary repos - since we won't support those arbitrary repos for things like retrieving WDL.